pkg/vsphere/vm/vm.go

```diff
@@ -281,7 +284,37 @@ func (vm *VirtualMachine) DeleteExceptDisks(ctx context.Context) (*object.Task,
 		return nil, err
 	}
 
-	return vm.Destroy(op)
+	info, err := vm.WaitForResult(op, func(ctx context.Context) (tasks.Task, error) {
+		return vm.Destroy(ctx)
+	})
+
+	if err == nil {
+		return info, nil
+	}
+
+	if isMethodDisabledError(ctx, err) {
+		vm.EnableDestroy(ctx)
```

Comment:
If we fail calling `EnableDestroy` then we will not be able to delete the VM below and it's worthy of a custom error message and Warning logging - it requires user intervention to clean up at this point.

Reply:
If EnableDestroy failed, a warning message is already logged by EnableDestroy function, but we still attempt to destroy the VM after. We should not attempt to destroy the VM if EnableDestroy failed, and we should return the error back to the caller for retry. Fixing it.